app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/DiscoveryBottomSheetDialog.kt

```diff
@@ -54,10 +61,38 @@ abstract class DiscoveryBottomSheetDialog : InformationBottomSheetDialog() {
         }
     }
 
+    private fun setIllustration() = with(binding) {
+        when (val illustration = illustration) {
+            is Illustration.Static -> infoIllustration.apply {
+                isVisible = true
+                setBackgroundResource(illustration.resId)
+            }
+            is Illustration.Animated -> infoAnimation.apply {
+                isVisible = true
+
+                val config = Config.Builder()
+                    .autoplay(true)
+                    .source(DotLottieSource.Res(illustration.resId))
+                    .loop(true)
+                    .playMode(Mode.REVERSE)
+                    .useFrameInterpolation(true)
+                    .build()
+
+                infoAnimation.load(config)
+            }
+        }
+    }
+
     abstract fun onPositiveButtonClicked()
 
     override fun onCancel(dialog: DialogInterface) {
         trackMatomoWithCategory(MatomoName.DiscoverLater)
         super.onCancel(dialog)
     }
 }
+
+// TODO: Move this somewhere else?
+sealed interface Illustration {
+    data class Static(@DrawableRes val resId: Int) : Illustration
+    data class Animated(@RawRes val resId: Int) : Illustration
+}
```

Comment:
I'd move it inside of DiscoveryBottomSheet so it's accessed as such from the outside: `DiscoveryBottomSheet.Illustration` otherwise we need to name it more clearly like `DiscoveryIllustration` but it would make it available everywhere for not real reason imo. Also, accessing a "discover illustration" from inside a "discovery bottom sheet" is a bit redundant